### ISSUE TITLE
Is it possible to have other Juicebox normalisation method small bash script for STARE

### ISSUE THREAD

**Issue by kyliuaefro** [SchulzLab/STARE#12]
Hello,
Is it possible to have other Juicebox normalisation method small bash script for STARE?
`./Code/Juicebox_KR_normalization.sh -j <path_juicer_jar_file> -h <hic_path> -d <out_path> -c <chromosomes> -b <bin_size>
`

I do not have an example to shown but i know my public data (U-87) have a small part of chromosome truncated in Chr9 (not 100% sure sorry) from this link : https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM4417604

However, when i use the build in bash in STARE, i lost the whole contact file in chr9. Me and my labmmates discussed about it and think it is the problem of the normalization method from Juicebox. 

So i want to ask if you guys can have any suggestion or help me how to fix this?/ Have option in the bash code to allow another normalization method? i am not quite good at coding. Hope what i said did not confuse you guys. 

Thank you very much

David

**Comment by DennisHeck** [comment]
Hey David,

We have seen this issue before. Knight-Ruiz normalisation sometimes doesn't converge, especially on short chromosomes, which leads to empty files by juicebox. 

I added another small script under https://github.com/SchulzLab/STARE/blob/main/Code/Juicebox_normalization.sh. It's the same as before but takes an additional option -n for defining the normalisation. The methods supported by Juicebox should be KR, VC and VC_SQRT. I tested VC for chr9 for the HiC file you linked, and it worked. 

Generally, I'm not sure how well VC performs compared to KR, but intuitively I would say that as long as the contact frequencies are comparable across interactions in a window it should be fine.

Let me know if there are any issues. Best wishes,
Dennis
